GVFS/GVFS.Common/ProductUpgrader.cs

```diff
@@ -83,37 +87,71 @@ protected ProductUpgrader()
             out ProductUpgrader newUpgrader,
             out string error)
         {
-            // Prefer to use the NuGet upgrader if it is configured. If the NuGet upgrader is not configured,
-            // then try to use the GitHubUpgrader.
-            if (NuGetUpgrader.TryCreate(tracer, fileSystem, gvfsConfig, credentialStore, dryRun, noVerify, out NuGetUpgrader nuGetUpgrader, out bool isConfigured, out error))
+            Dictionary<string, string> entries;
+            if (!gvfsConfig.TryGetAllConfig(out entries, out error))
             {
-                // We were successfully able to load a NuGetUpgrader - use that.
-                newUpgrader = nuGetUpgrader;
-                return true;
+                newUpgrader = null;
+                return false;
+            }
+
+            bool containsUpgradeFeedUrl = entries.ContainsKey(GVFSConstants.LocalGVFSConfig.UpgradeFeedUrl);
+            bool containsUpgradePackageName = entries.ContainsKey(GVFSConstants.LocalGVFSConfig.UpgradeFeedPackageName);
+            bool containsOrgInfoServerUrl = entries.ContainsKey(GVFSConstants.LocalGVFSConfig.OrgInfoServerUrl);
+
+            if (containsUpgradeFeedUrl || containsUpgradePackageName)
+            {
+                // We are configured for NuGet - determine if we are using OrgNuGetUpgrader or not
+                if (containsOrgInfoServerUrl)
+                {
+                    if (OrgNuGetUpgrader.TryCreate(
+                        tracer,
+                        fileSystem,
+                        gvfsConfig,
+                        new HttpClient(),
+                        credentialStore,
+                        dryRun,
+                        noVerify,
+                        out OrgNuGetUpgrader orgNuGetUpgrader,
+                        out error))
+                    {
+                        // We were successfully able to load a NuGetUpgrader - use that.
+                        newUpgrader = orgNuGetUpgrader;
+                        return true;
+                    }
+                    else
+                    {
+                        tracer.RelatedError($"{nameof(TryCreateUpgrader)}: Could not create upgrader. {error}");
+                        newUpgrader = null;
+                        return false;
+                    }
+                }
+                else
+                {
+                    if (NuGetUpgrader.TryCreate(tracer, fileSystem, gvfsConfig, credentialStore, dryRun, noVerify, out NuGetUpgrader nuGetUpgrader, out bool isConfigured, out error))
+                    {
+                        // We were successfully able to load a NuGetUpgrader - use that.
+                        newUpgrader = nuGetUpgrader;
+                        return true;
+                    }
+                    else
+                    {
+                        tracer.RelatedError($"{nameof(TryCreateUpgrader)}: Could not create upgrader. {error}");

```

Comment:
Make this `RelatedError` text and the text above (on line 123) different to guarantee we know which error case was hit?